python/mxnet/gluon/block.py

```diff
@@ -1083,6 +1083,10 @@ def _build_cache(self, *args):
 
             self._cached_op_args.append(triple)
 
+        for i in range(len(self._flags) - 1, -1, -1):
```

Comment:
@waytrue17 why are we not using `for i in range(len(self._flags)):` here?

Reply:
We want to add/updated the indices. The current implementation is to check and remove the existing indices and then append the new ones. Using `for i in range(len(self._flags))` may cause some index issue when removing items in-place